File: OnlineDiaryClient/src/app/user/models/tasks.ts
export class Task{
    title:string="";
    tags:string="";
    startDate:Date=null;
    endDate:Date=null;
    description:string="";
    assignedTo:Array<string>=[];
    assignedBy:string="";
    createdOn:Date=null;
    updatedBy:Date=null;
    createdBy:string="";
    isActive:boolean=true;
    isCompleted:boolean=false;
    status:string="";
}

File: OnlineDiaryServer/OD.Entities/Tasks.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Text;

namespace OD.Entities
{
    public partial class Tasks : IEntity
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [BsonIgnoreIfDefault]
        public string Id { get; set; }
        public string Title { get; set; }
        public List<string> Tags { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public List<string> AssignedTo { get; set; }
        public string AssignedBy { get; set; }
        public string Description { get; set; }
        public DateTime? CreatedOn { get; set; }
        public string CreatedBy { get; set; }
        public bool IsActive { get; set; }
        public bool IsCompleted { get; set; }
        public string Status { get; set; }
        public string UpdatedBy { get; set; }
    }
}
